src/panel/applets/status/PowerIndicator.vala

```diff
@@ -192,6 +204,53 @@ public class PowerIndicator : Gtk.Bin {
 		button.clicked.connect(open_power_settings);
 		button.get_child().set_halign(Gtk.Align.START);
 		box.pack_start(button, false, false, 0);
+
+		power_profiles = new PowerProfiles();
+		
+		power_profiles.proxy_connected.connect(() => {
```

Comment:
If PowerProfiles is invalid for any reason e.g. bus issues, powerprofiles are not applicable to the desktop the proxy_connected would be invalid?

Probably should test this - if there is a failure then powerprofiles in the indicator should probably be hidden

Reply:
proxy_connected is a signal emitted by PowerProfiles.vala- if it doesn't get the bus for whatever reason, the callback simply shouldn't ever be called. There may be a better way to structure it to make that clear, though.

Reply:
^ update: yep, it renders regardless of the existence of the bus for whatever reason